File: Stack/Stack_by_array/stackarray.py
class stack :
    def __init__(self):
        self.top = -1 
        self.list = []

    def isEmpty(self):
        return self.top == -1 
    
    def isFull(self):
        return self.top == len(self.list)
    def push(self, data):
        if(self.isFull()):
            print("Stack is overflow !!!") 
            return 
        self.list.append(data) 
        self.top = self.top + 1 
    def pop(self):
        if(self.isEmpty()):
            print("Stack is underflow !!!")
            return 
        x = self.list.pop(self.top)
        return x 

    def peek(self):
        return self.list[self.top] 
    def display(self):
        for i in self.list:
            print(i, end = " ")
s = stack()

s.push(10) 
s.push(20)
s.push(30)
s.push(40) 
print("Print item : ")
s.display() 
print("\n") 
print("Item popped from stack : ",s.pop()) 

File: Stack/Stack_by_array/stack_array.cpp
#include <iostream> 
using namespace std; 
#define MAX 1000 
class Stack{
    int top ; 

    public :
        int a[MAX] ;
        Stack () {top = -1 ;}
        bool push( int x) ; 
        int pop() ; 
        int peek() ; 
        bool isEmpty() ;  
}; 

bool Stack :: push(int x){
    if(top >= MAX-1){
        cout<<"Stack overflow ." ;
        return false ;  
    }else {
        a[++top] = x ; 
        cout<<x<<" pushed into stack \n" ; 
        return true ; 
    }
}

int Stack :: pop(){
    if(top < 0 ){
        cout<<"Stack overflow ." ; 
        return 0 ; 
    }else {
        int x = a[top--] ; 
        return x ; 
    }
}

int Stack :: peek(){
    if(top < 0){
        cout<<"Stack is empty ." ; 
        return 0 ; 
    }else{
        int x = a[top] ; 
        return x ; 
    }
}

bool Stack :: isEmpty(){
    return top < 0 ; 
}

int main(){
    class Stack s ; 
    s.push(10) ; 
    s.push(20) ; 
    s.push(30) ; 
    cout<<s.pop()<< "Popped from Stack \n" ; 
    cout<<"Top element is : "<<s.peek() << endl; 

    cout<<"Element present in Stack : " ;
    while(!s.isEmpty()){
        cout<<s.peek()<<" " ; 
        s.pop() ; 
    }

    return 0 ; 

}